Linked sliders e2e

AnOctopus opened the pull request:
Adding some testing of the linked sliders example app. Ideally more will be added. This PR is partly to check if the potentially fragile conditions in the test code will work in CI.

AnOctopus committed:
Add e2e regression test for linked sliders

AnOctopus committed:
More linked_sliders tests, longer timeouts

AnOctopus committed:
Wrap e2e test in check for streamlit running

Otherwise the script will crash when the integration tests run it bare.

vdonato commented on e2e/scripts/linked_sliders.py:
```diff
@@ -0,0 +1,55 @@
+
+import streamlit as st
+
+if st._is_running_with_streamlit:
```
nit: can get rid of this check now

vdonato commented on e2e/scripts/linked_sliders.py:
```diff
@@ -0,0 +1,55 @@
+        "Fahrenheit",
+        min_value=to_fahrenheit(MIN_CELSIUS),
+        max_value=to_fahrenheit(MAX_CELSIUS),
+        value=to_fahrenheit(state.celsius),
```
nit: this will print a warning since it sets a slider value while also setting the value of the slider via session_state within a callback

vdonato commented on e2e/specs/linked_sliders.spec.js:
```diff
@@ -0,0 +1,90 @@
+      .eq(0)
+      .parent()
+      .click();
+    cy.wait(1000);
```
is the `cy.wait` necessary? I think cypress automatically waits until the `.should("have.text", ...)` is true before proceeding, so the lines below should work without this

AnOctopus replied:
Yes. When running this locally, I found that it consistently failed due to the page updating after the click.

vdonato commented on e2e/specs/linked_sliders.spec.js:
```diff
@@ -0,0 +1,90 @@
+    cy.get('.stSlider [role="slider"]')
+      .eq(1)
+      .parent()
+      .click(698, 0, { force: true });
```
is there any significance to the 698 here?

AnOctopus replied:
It is the width of the slider. I tried using `.click("right")` but it clicked just short of the very end of the slider, so the values weren't the max temperatures that was expected.

AnOctopus committed:
Fix warning and always run code in linked slider e2e

AnOctopus committed:
Add comments for weird parts of linked sliders e2e

AnOctopus committed:
Merge branch 'feature/session-state' into linked-sliders-e2e

AnOctopus committed:
Readd check for running in streamlit in e2e

Because lookups raise an exception if the key is missing, the empty
SessionState instance we use when there is no session just results in
those exceptions, so we still have to wrap the tests for bare execution
for now.
